soc/arm/nxp_s32/s32k/soc.c

```diff
@@ -72,6 +76,10 @@ static int soc_init(void)
 
 	OsIf_Init(NULL);
 
+#ifdef CONFIG_NXP_S32_PM
+	nxp_s32_power_init();
```

Comment:
why call the `power_init` function here, instead of using `SYS_INIT` in the `power_soc.c` file? I'd prefer `SYS_INIT` because this avoids the need to expose the `power_init` function via a header file

Reply:
The reason was to run it after SoC init and before clock control, but there's no need, it can be done with init priorities. Updated to use `SYS_INIT`.

Reply:
HAL PR merged. Please update the `west.yml` file